Automattic/jetpack
# Super Cache & Boost: detect WP_CACHE when it's set using "const" in wp-config.php

I've never seen this as a complaint while supporting WP Super Cache but it's possible for WP_CACHE to be defined using "const" by some other plugin so we should detect that and not try to add a define() to set it again.

Fixes #36358

## Proposed changes:

* Modified the Boost regex to check for "const WP_CACHE".
* Added the same regex to WPSC so it checks for both define and const, before doing any updates to wp-config.php


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





* Add `const WP_CACHE = true;` to wp-config.php
* Load the settings page for Jetpack Boost and enable the Page Cache module. There shouldn't be any errors in your error_log related to WP_CACHE.
* Load the settings page for WPSC, and make sure it's active. Again, there shouldn't be any WP_CACHE errors in the error_log.
